packages/runtime/container-runtime/src/dataStore.ts

```diff
@@ -128,6 +135,13 @@ class DataStore implements IDataStore {
     ) { }
     public get IFluidRouter() { return this.fluidDataStoreChannel; }
 
+    /**
+     * @deprecated - This will be removed once https://github.com/microsoft/FluidFramework/issues/9127 is fixed.
+     */
+    public bindToContext() {
```

Comment:
why expose this, and add the new interface? i would just do one or the other

Reply:
I thought not making this public would result in a compiler error when some one casts to the interface and calls it.

Reply:
no. they may need to cast thru any, or unknown, but thats fine

Reply:
i would have the above reach down rather than expose here

Reply:
`interface AccessBindToContext extends IDataStore{ fluidDataStoreChannel?:{ bindToContext?(): void;}}`

Reply:
the user will do something this like
```
const ds = createDatastore();
const dsWithBind = ds as any as IDataStoreWithBindToContext_Deprecated;
```
``

Reply:
Updated this. Can you please take a look.

Reply:
looks great! thanks!